packages/notebook-extension/src/index.ts

```diff
@@ -816,6 +816,14 @@ function activateWidgetFactory(
   translator: ITranslator,
   settingRegistry: ISettingRegistry | null
 ): NotebookWidgetFactory.IFactory {
+  const preferKernelOption = PageConfig.getOption('notebookStartsKernel');
+
+  // If the option is not set, assume `true`
+  const preferKernelValue: boolean =
+    preferKernelOption === ''
+      ? true
+      : preferKernelOption.toLowerCase() === 'true';
```

Comment:
I think this is clearer to tell it is true if not set or set to `true`.

```suggestion
  const preferKernelValue = preferKernelOption === '' || preferKernelOption.toLowerCase() === 'true';
```